internal/framework/generic_resource.go

```diff
@@ -236,8 +267,8 @@ func (r *GenericResource) ImportState(ctx context.Context, req resource.ImportSt
 const (
 	READINESS_TIMEOUT = 10 * time.Minute
 	DELETION_TIMEOUT  = 1 * time.Minute
-	POLLING_INTERVAL  = 5 * time.Second
-	FAILURE_THRESHOLD = 3
+	POLLING_INTERVAL  = 10 * time.Second
+	FAILURE_THRESHOLD = POLLING_INTERVAL + 1*time.Second
```

Comment:
For polling the threshold is effectively 2, but this change also increases the polling interval, so that we are actually waiting 20 seconds instead of 15 for the resource to change. There is also a small change in the way the polling loop works now where we always update the state before performing the first check.

Previously we were performing the first check based on the state read before doing `PUT`. So now we have one fewer stale `Failed` state that we can observe.

Reply:
Thanks for the explanation! I missed the fact that the polling interval is bumped.